src/lib/datepicker/datepicker-input.ts

```diff
@@ -56,14 +59,17 @@ export class MdDatepickerInput implements AfterContentInit, ControlValueAccessor
   /** The value of the input. */
   @Input()
   get value(): SimpleDate {
-    return this._value;
+    return this._locale.parseDate(this._elementRef.nativeElement.value);
   }
   set value(value: SimpleDate) {
-    this._value = this._locale.parseDate(value);
-    const stringValue = this._value == null ? '' : this._locale.formatDate(this._value);
-    this._renderer.setElementProperty(this._elementRef.nativeElement, 'value', stringValue);
+    let date = this._locale.parseDate(value);
+    let oldDate = this.value;
+    this._renderer.setElementProperty(this._elementRef.nativeElement, 'value',
+        date ? this._locale.formatDate(date) : '');
+    if (!SimpleDate.equals(oldDate, date)) {
+      this._valueChangeEmitter.emit(date);
```

Comment:
Yeah, I see how it's solving that problem. Just concerned that users will be confused because typically when you have a model to view change, you don't expect the view to model handler to fire as well.